src/ui/pages/work/partusage/PartUsageDialog.js

```diff
@@ -136,7 +136,21 @@ function PartUsageDialog(props) {
 
     let handleSave = () => {
         //Call the handle save from the parent
-        props.handleSave(partUsage, partUsageLine);
+        setLoading(true);
+        //Set the part usage Line
+        partUsage.transactionlines = [partUsageLine];
+        //Remove transaction info prop
+        delete partUsage.transactionInfo;
```

Comment:
This is used to prepare the `partUsage` properties to later pass onto the `createPartUsage` endpoint. As this variable is retrieved with `useState`, we should normally never modify it directly. As we do not wish to modify the state of this component here, merely make some changes before passing it to the endpoint, we should clone the `partUsage` object instead, and make the required changes before passing it to the endpoint.